### PR TITLE
Feature include api in card

### PR DESCRIPTION
**Pull request by Mmarvinn**

### CODE REVIEW

File: src/components/card.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 'use strict';
 
-export function createMovieCard() {
+import {getTopRatedFilms} from '../services/api.js';
+
+const BASE_IMG_URL = 'http://image.tmdb.org/t/p/original';
+
+export function createMovieCard(posterOfFilm, filmName, filmOverview) {
```
Comment (TarasProtsyshyn) [inline src/components/card.js]:
```js
fn({a : "a",  b: "dd",  c: "ss" })
```

```js 
createMovieCard(moviInfo);
```

File: src/pages/movieList.js [hunk 2]
```diff
@@ -2,16 +2,25 @@
...
+            const filmOverview = oneFilm[2];
 
-    const main = document.querySelector('main');
-    main.className = 'main--movie-list';
+            arrayWithCards.push(createMovieCard(posterOfFilm, filmName, filmOverview));
+        }
+    
+            renderPageLayout(arrayWithCards);
+
+        const main = document.querySelector('main');
```
Comment (TarasProtsyshyn) [inline src/pages/movieList.js]: @Mmarvinn better to have opportunity to pass a class name to `renderPageLayout` fn

File: src/pages/movieList.js [hunk 3]
```diff
@@ -2,16 +2,25 @@
...
 export function createMovieList() {
     const arrayWithCards = [];
 
-    for (let i = 0; i < 20; i++) {
-        arrayWithCards.push(createMovieCard());
-    }
+    getTopRatedFilms().then(result => {
 
-    renderPageLayout(arrayWithCards);
+        for (let i = 0; i < result.length; i++) {
```
Comment (TarasProtsyshyn) [inline src/pages/movieList.js]: pls, use the here one of array's methods

File: src/services/api.js [hunk 4]
```diff
@@ -9,27 +9,30 @@ export async function getTopRatedFilms() {
     let response = await fetch(`${BASE_URL}/movie/popular?api_key=${API_KEY}&language=en-US&page=1`);
     let json = await response.json();
 
-    let propOfFilm = [];
+    const propOfFilm = [];
```
Comment (TarasProtsyshyn) [inline src/services/api.js]: @Mmarvinn  pls remove it, it is redundant

File: src/components/card.js (hunk 1 above)
Comment (Mmarvinn) [reply src/components/card.js]: fix

File: src/pages/movieList.js (hunk 2 above)
Comment (Mmarvinn) [reply src/pages/movieList.js]: fix

File: src/pages/movieList.js (hunk 3 above)
Comment (Mmarvinn) [reply src/pages/movieList.js]: fixed

File: src/services/api.js (hunk 4 above)
Comment (Mmarvinn) [reply src/services/api.js]: fixed

File: src/pages/movieDetailing.js [hunk 5]
```diff
@@ -4,13 +4,13 @@ import {renderPageLayout} from '../layouts/createLayout.js';
 
 export function createMovieInfo() {
     const movieDetailBlock = createStructurePageMovieDetailing();
- 
-    renderPageLayout(movieDetailBlock);
+    const mainClassName = 'main--movie-info';
```
Comment (TarasProtsyshyn) [inline src/pages/movieDetailing.js]: @Mmarvinn Why do you need, here new variables? Pls pass this className direct to the function
```js
renderPageLayout(movieDetailBlock, 'main--movie-info');
```

File: src/pages/movieList.js [hunk 6]
```diff
@@ -2,16 +2,24 @@
...
 export function createMovieList() {
     const arrayWithCards = [];
 
-    for (let i = 0; i < 20; i++) {
-        arrayWithCards.push(createMovieCard());
-    }
+    getTopRatedFilms().then(result => {
+        
+        result.results.forEach(element => {
+            const movieInfo = {};
```
Comment (TarasProtsyshyn) [inline src/pages/movieList.js]: @Mmarvinn You do not need this separate variable for passing parameters in the function. You can just write like this 
```js
arrayWithCards.push(createMovieCard({
  posterOfFilm: element.poster_path,
  filmName:  element.original_title,
  filmOverview:   element.overview,
}));
```

File: src/services/api.js [hunk 7]
```diff
@@ -1,36 +1,12 @@
 'use strict';
 
 const BASE_URL = 'https://api.themoviedb.org/3';
-const BASE_IMG_URL = 'http://image.tmdb.org/t/p/';
 const API_KEY = '1be31873de9aa764d36bafbd53752fba';
 
 export async function getTopRatedFilms() {
 
     let response = await fetch(`${BASE_URL}/movie/popular?api_key=${API_KEY}&language=en-US&page=1`);
     let json = await response.json();
```
Comment (TarasProtsyshyn) [inline src/services/api.js]: @Mmarvinn  Actually `response.json();` returning a `promise` that resolves to a JavaScript object. So will be  better to rename it to `data`

File: src/pages/movieList.js [hunk 8]
```diff
@@ -2,16 +2,24 @@
...
 import {renderPageLayout} from '../layouts/createLayout.js';
+import {getTopRatedFilms} from '../services/api.js';
 
 export function createMovieList() {
     const arrayWithCards = [];
 
-    for (let i = 0; i < 20; i++) {
-        arrayWithCards.push(createMovieCard());
-    }
+    getTopRatedFilms().then(result => {
```
Comment (TarasProtsyshyn) [inline src/pages/movieList.js]: @Mmarvinn  pls rename `result` to `data`

**Review by TarasProtsyshyn — COMMENTED** [review]
@Mmarvinn What about handling a loading state and error state?

**Comment by TarasProtsyshyn** [comment]
@Mmarvinn  one more remark.  This is not an excellent message `fix class names, use arr methods, and use obj in arguments` for commit. If you have a counting or `and` in your commit message that indicates that it will be better to split this commit into e few smaller ones. Pls avoided it in the future.

File: src/pages/movieDetailing.js (hunk 5 above)
Comment (Mmarvinn) [reply src/pages/movieDetailing.js]: fix

File: src/services/api.js (hunk 7 above)
Comment (Mmarvinn) [reply src/services/api.js]: thx.
fix

File: src/pages/movieList.js (hunk 8 above)
Comment (Mmarvinn) [reply src/pages/movieList.js]: fix

File: src/pages/movieList.js (hunk 6 above)
Comment (Mmarvinn) [reply src/pages/movieList.js]: fix

File: main.css [hunk 9]
```diff
@@ -91,21 +101,150 @@ header {
...
+
+.movie-container--content {
+    z-index: 1;
+    font-size: 1.25rem;
+    font-family: cursive;
+    grid-column: span 3;
+    margin: 0 30px 0 0;
+}
+
+.movie-container--content--wrapper-rating {
```
Comment (taras-protsyshyn) [inline main.css]: @Mmarvinn The wrong class name, pls follow BEM methodology

File: main.css [hunk 10]
```diff
@@ -91,21 +101,150 @@ header {
...
+    margin-top: 30px;
+    margin-bottom: 30px;
+}
+
+.movie-container--content--wrapper-rating--rating-text {
```
Comment (taras-protsyshyn) [inline main.css]: @Mmarvinn The wrong class name, pls follow BEM methodology

File: main.css [hunk 11]
```diff
@@ -91,21 +101,150 @@ header {
...
+    font-size: 1.5rem;
+}
+
+.movie-container--content--wrapper-rating--wrapper-rating-icon {
```
Comment (taras-protsyshyn) [inline main.css]: @Mmarvinn The wrong class name, pls follow BEM methodology

File: main.css [hunk 12]
```diff
@@ -91,21 +101,150 @@ header {
...
+}
+
+.movie-container--content--wrapper-about-movie {
+    display: flex;
+    flex-wrap: nowrap;
+}
+
+.movie-container--content--wrapper-about-movie--release-date,
+.movie-container--content--wrapper-about-movie--movie-genre,
+.movie-container--content--wrapper-about-movie--movie-duration {
```
Comment (taras-protsyshyn) [inline main.css]: @Mmarvinn The wrong class name, pls follow BEM methodology

File: main.css [hunk 13]
```diff
@@ -91,21 +101,150 @@ header {
...
+    color: red;
+    overflow: hidden;
+    letter-spacing: -4px;
+}
+
+.release-year {
+    margin-left: 20px;
+}
+
+.movie-container--content--wrapper-about-movie {
```
Comment (taras-protsyshyn) [inline main.css]: @Mmarvinn The wrong class name, pls follow BEM methodology

File: src/components/clearBody.js [hunk 14]
```diff
@@ -0,0 +1,9 @@
+export const clearBody = () => {
```
Comment (taras-protsyshyn) [inline src/components/clearBody.js]: @Mmarvinn This is not a component, I think it a utility

File: src/components/displayError.js [hunk 15]
```diff
@@ -0,0 +1,15 @@
+import { createHeader } from '../layouts/createLayout.js';
+import { createFooter } from '../layouts/createLayout.js';
+import { clearBody } from '../components/clearBody.js';
+
+export function displayError(error) {
```
Comment (taras-protsyshyn) [inline src/components/displayError.js:3]: @Mmarvinn It's not needed you can use your `renderPageLayout` function. Just like this 
```js
renderPageLayout(`Error status: ${error.status}`)
```

File: src/components/displayLoading.js [hunk 16]
```diff
@@ -0,0 +1,19 @@
+import { createHeader } from '../layouts/createLayout.js';
+import { createFooter } from '../layouts/createLayout.js';
+import { clearBody } from '../components/clearBody.js';
+
+export function displayLoading() {
```
Comment (taras-protsyshyn) [inline src/components/displayLoading.js:3]: @Mmarvinn The same as previous

File: src/components/clearBody.js (hunk 14 above)
Comment (Mmarvinn) [reply src/components/clearBody.js]: fix

File: src/components/displayError.js (hunk 15 above)
Comment (Mmarvinn) [reply src/components/displayError.js:3]: fix

File: src/components/displayLoading.js (hunk 16 above)
Comment (Mmarvinn) [reply src/components/displayLoading.js:3]: fix

File: main.css (hunk 9 above)
Comment (Mmarvinn) [reply main.css]: fix

File: main.css (hunk 10 above)
Comment (Mmarvinn) [reply main.css]: fix

File: main.css (hunk 11 above)
Comment (Mmarvinn) [reply main.css]: fix

File: main.css (hunk 12 above)
Comment (Mmarvinn) [reply main.css]: fix

File: main.css (hunk 13 above)
Comment (Mmarvinn) [reply main.css]: fix

File: main.css [hunk 17]
```diff
@@ -91,21 +101,150 @@ header {
...
+    height: 100%;
+    content: "☆☆☆☆☆";
+    position: absolute;
+    top: 0;
+    left: 0;
+    font-size: 40px;
+    letter-spacing: -4px;
+}
+
+.rating-body__rating-line {
```
Comment (taras-protsyshyn) [inline main.css]: @Mmarvinn `rating-line` - not sound like a modifier

File: src/services/getFilmId.js [hunk 18]
```diff
@@ -0,0 +1,3 @@
+export function getFilmId() {
```
Comment (taras-protsyshyn) [inline src/services/getFilmId.js]: @Mmarvinn I would be put this function to `utils`, it is too small for `services`

File: src/utility/clearBody.js [hunk 19]
```diff
@@ -0,0 +1,9 @@
+export const clearBody = () => {
```
Comment (taras-protsyshyn) [inline src/utility/clearBody.js]: @Mmarvinn Pls rename folder from `utility` to `utils`

**Review by taras-protsyshyn — COMMENTED** [review]
@Mmarvinn why do you need these images images/details-page.jpg images/main-page.jpg?

File: main.css (hunk 17 above)
Comment (Mmarvinn) [reply main.css]: fix

File: src/services/getFilmId.js (hunk 18 above)
Comment (Mmarvinn) [reply src/services/getFilmId.js]: fix

File: src/utility/clearBody.js (hunk 19 above)
Comment (Mmarvinn) [reply src/utility/clearBody.js]: fix

**Pull request merged**